fairlearn/reductions/_grid_search/grid_search.py

```diff
@@ -198,7 +198,7 @@ def predict_fct(X): return current_estimator.predict(X)
         if self.selection_rule == TRADEOFF_OPTIMIZATION:
             def loss_fct(i):
                 return self.objective_weight * self.objectives_[i] + \
-                    self.constraint_weight * self.gammas_[i].max()
+                    self.constraint_weight * self.gammas_[grid.columns[i]].max()
```

Comment:
I'm honestly a bit confused here. Above it says `for i in grid.columns` which should iterate over the column names, right? So how does `grid.columns[i]` fix this?

`grid` doesn't have a specified input type in the API reference so I'm a bit worried we're breaking this in case people pass a `np.ndarray`. Perhaps I'm misinterpreting this but I think so far we assumed `grid` to be a list of columns. I don't quite know for sure since I didn't write the original code and re-reading it isn't making things super obvious either. @riedgar-ms  @MiroDudik

Reply:
Line 162 says `for i in grid.columns`, but the `i` on line 201 refers to the argument of the `loss_fct` which comes from the list comprehension on line 202 `for i in range(len(self.objectives_))`